packages/nextjs/components/scaffold-eth/Contract/ContractUI.tsx

```diff
@@ -1,82 +1,194 @@
+              <div className="z-10">
+                <div className="bg-white rounded-2xl shadow-xl border flex flex-col mt-10 relative">
+                  <div className="h-[5rem] w-[5.5rem] bg-secondary absolute self-start rounded-[22px] -top-[38px] -left-[1px] -z-10 py-[0.65rem] shadow-lg shadow-base-300">
+                    <div className="flex items-center justify-center space-x-2">
+                      <p className="my-0 text-sm font-bold">Write</p>
+                    </div>
+                  </div>
+                  <div className="divide-y divide-base-300 p-5">
+                    <ContractWriteMethods
+                      deployedContractData={{ address: initialContractData.address, abi }}
+                      onChange={triggerRefreshDisplayVariables}
+                      removeMethod={removeMethod}
+                    />
+                  </div>
+                </div>
               </div>
             </div>
-          </div>
-          <div className="z-10">
-            <div className="bg-white rounded-2xl shadow-xl border flex flex-col mt-10 relative">
-              <div className="h-[5rem] w-[5.5rem] bg-secondary absolute self-start rounded-[22px] -top-[38px] -left-[1px] -z-10 py-[0.65rem] shadow-lg shadow-base-300">
-                <div className="flex items-center justify-center space-x-2">
-                  <p className="my-0 text-sm font-bold">Write</p>
+
+            <div className="col-span-3 flex flex-col mt-10">
+              <div className="bg-white border shadow-xl rounded-2xl px-6 lg:px-8 mb-6 space-y-1 py-4">
+                <div className="flex">
+                  <div className="flex flex-col gap-1">
+                    <span className="font-bold">Contract Data</span>
+                    <Address address={initialContractData.address} />
+                    <div className="flex items-center gap-1">
+                      <span className="text-sm font-bold">Balance:</span>
+                      <Balance address={initialContractData.address} className="h-1.5 min-h-[0.375rem] px-0" />
+                    </div>
+                  </div>
                 </div>
+                {mainNetwork && (
+                  <p className="my-0 text-sm">
+                    <span className="font-bold">Network</span>:{" "}
+                    <span style={{ color: networkColor }}>{mainNetwork.name}</span>
+                  </p>
+                )}
               </div>
-              <div className="divide-y divide-base-300 p-5">
-                <ContractWriteMethods
-                  deployedContractData={deployedContractData}
-                  onChange={triggerRefreshDisplayVariables}
+              <div className="bg-white shadow-xl rounded-2xl px-6 lg:px-8 py-4">
+                <ContractVariables
+                  refreshDisplayVariables={refreshDisplayVariables}
+                  deployedContractData={{ address: initialContractData.address, abi }}
```

Comment:
Agree with Pablo! 
1. Lets pass `initialContractData.abi` to since `ContractVariables` component already has logic to filter ContractVariables and show them
2. Lets keep default `abi` state as `[]` array

Reply:
Thanks! I focused on other things too much and did not even notice that I overrided abi state var in useEffect. Fixed 🫡